zerver/actions/message_send.py

```diff
@@ -843,7 +843,7 @@ def do_send_messages(
     email_gateway: bool = False,
     scheduled_message_to_self: bool = False,
     mark_as_read: Sequence[int] = [],
-) -> List[int]:
+) -> List[Tuple[int, Optional[int]]]:
```

Comment:
Rather than returning a tuple, let's just make this return a `SentMessageResult` data class that this can return; it'll be much more extensible in the future and save adding `ignored` parameters in various places, which is always a bit messy..

Reply:
Done